application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,152 @@
+            .sendMessageEmbeds(new EmbedBuilder().setTitle(title)
+                .setDescription(description)
+                .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                .setTimestamp(timestamp)
+                .setColor(EMBED_COLOR)
+                .build());
+
+        for (Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContent(), attachment.getName());
+        }
+        message.queue();
+    }
+
+    /**
+     * logs an entry in the mod audit log channel.
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user who triggered the action
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     *
+     * @param attachment an attachment that'll be added to the message
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild,
+            @NotNull Attachment attachment) {
+        writeModAuditLog(title, description, author, timestamp, guild, List.of(attachment));
+    }
+
+    /**
+     * logs an entry in the mod audit log channel.
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user who triggered the action
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild) {
+        writeModAuditLog(title, description, author, timestamp, guild, List.of());
+    }
+
+    /**
+     * used to get the mod audit log channel.
+     *
+     * @param guild the current guild
+     */
+    public static Optional<TextChannel> getModAuditLogChannel(@NotNull Guild guild) {
+        Optional<TextChannel> channel =
+                guild.getTextChannelCache().stream().filter(isAuditLogChannel).findAny();
+        if (channel.isEmpty()) {
+            logger.warn(
+                    "Unable to log moderation events, did not find a mod audit log channel matching the configured pattern '{}' for guild '{}'",
+                    Config.getInstance().getModAuditLogChannelPattern(), guild.getName());
+        }
+        return channel;
+    }
+
+    /**
+     * used to add a file to a message without having an actual file.
```

Comment:
Same issue as before with the way you worded this. Instead, describe what the class is. So `"Represents attachment to messages, as for example used by {@link Message#addFile(...)}.`

Reply:
done.